.github/workflows/prepare-release.yaml

```diff
@@ -0,0 +1,82 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Prepare Release
+
+on:
+  push:
+    branches:
+      - release/*
+
+jobs:
+  prepare-release:
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      PYTHONPATH: /opt/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/site-packages
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Checkout repo
```

Comment:
Even fetch-depth = 0 does not help and git log says tags are bad revision. As a workaround, clone the repo with git, checkout to a branch, run prepare_release script and then copy src back to the repo checkout in Checkout code step.